src/typegoose.ts

```diff
@@ -107,13 +107,25 @@ export function buildSchema<T, U extends AnyParamConstructor<T>>(cl: U, options?
   let sch: mongoose.Schema<U>;
   /** Parent Constructor */
   let parentCtor = Object.getPrototypeOf(cl.prototype).constructor;
+  const parentClasses: AnyParamConstructor<T>[] = [];
+
   // iterate trough all parents
   while (parentCtor?.name !== 'Object') {
-    // extend schema
-    sch = _buildSchema(parentCtor, sch, mergedOptions, false);
+    // add parent class to the beginning of the array if not added yet.
+    if (!parentClasses.find(cls => cls.name === parentCtor.name)) {
```

Comment:
why is that needed?
if this is really needed, please replace `constructor.name` with `getName(constructor)`

Reply:
Just checked and actually it's not needed, so will remove the condition.